autoload/sj/go.vim

```diff
@@ -52,9 +52,62 @@ function! sj#go#SplitStruct()
 endfunction
 
 function! sj#go#JoinStruct()
+  return s:joinStructOrFunc('{', '}')
+endfunction
+
+function! sj#go#SplitFunc()
+  let line = getline('.')
+  if line !~ '^func '
+    return 0
+  endif
+
+  let [start, end] = s:locateFuncBracesOnLine(line)
+  if start < 0 && end < 0
+    return 0
+  endif
+
+  let parsed = sj#ParseJsonObjectBody(start + 1, end - 1)
+
+  let arg_groups = []
+  let typed_arg_group = ''
+  for elem in parsed
+    if match(elem, '\s\+') != -1
+      let typed_arg_group .= elem
+      call add(arg_groups, typed_arg_group)
+      let typed_arg_group = ''
+    else
+      " not typed here, add to group
+      let typed_arg_group .= elem . ', '
+    endif
+  endfor
```

Comment:
Actually, nevermind, I get it now. The logic merges arguments like `a, b int,` into one item, so they're on the same line. That does make sense. It would be nice to put a short comment above to describe it, something like "keep `a, b int` variables on the same line". Or extract it to a helper function. As you say, some people might prefer `a int, b int` on separate lines, and that one could be a separate helper function (if it's requested).